Dependencies/napi/napi-jsi/include/napi/napi-inl.h

```diff
@@ -2181,13 +2181,40 @@ ObjectWrap<T>::DefineClass(napi_env env,
     descriptor.setProperty(rt, "configurable", jsi::Value{(p.attributes & napi_configurable) != 0});
     descriptor.setProperty(rt, "enumerable", jsi::Value{(p.attributes & napi_enumerable) != 0});
 
+    jsi::Object& owner = (p.staticVoidMethod || p.staticMethod || p.staticGetter || p.staticSetter || !p.staticValue.IsEmpty()) ?
+        constructor :
+        prototype;
+
     if (p.staticVoidMethod != nullptr) {
-      //throw std::runtime_error{"TODO"};
+      descriptor.setProperty(rt, "value", jsi::Function::createFromHostFunction(rt, name, 0,
+          [env, method{p.staticVoidMethod}, data{p.data}](jsi::Runtime& /*rt*/, const jsi::Value& thisVal, const jsi::Value* args, size_t count) -> jsi::Value {
+            (*method)({env, thisVal, args, count, nullptr, data});
+            return {};
+      }));
     } else if (p.staticMethod != nullptr) {
-      //throw std::runtime_error{"TODO"};
+      descriptor.setProperty(rt, "value", jsi::Function::createFromHostFunction(rt, name, 0,
+          [env, method{p.staticMethod}, data{p.data}](jsi::Runtime& /*rt*/, const jsi::Value& thisVal, const jsi::Value* args, size_t count) -> jsi::Value {
+            (*method)({env, thisVal, args, count, nullptr, data});
+            return {};
+      }));
     } else if (p.staticGetter != nullptr || p.staticSetter != nullptr) {
-      //throw std::runtime_error{"TODO"};
+      if (p.staticGetter != nullptr)
+      {
+          descriptor.setProperty(rt, "get", jsi::Function::createFromHostFunction(rt, name, 0,
+            [env, getter{ p.staticGetter }, data{ p.data }](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)-> jsi::Value {
+              return { rt, (*getter)({env, thisVal, args, count, nullptr, data}) };
```

Comment:
nit: inconsistent spacing
```suggestion
            [env, getter{p.staticGetter}, data{p.data}](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)-> jsi::Value {
              return {rt, (*getter)({env, thisVal, args, count, nullptr, data})};
```